oci/archive/oci_transport.go

```diff
@@ -0,0 +1,179 @@
+		tag = reference[sep+1:]
+	}
+	return NewReference(file, tag)
+}
+
+var refRegexp = regexp.MustCompile(`^([A-Za-z0-9._-]+)+$`)
+
+// NewReference returns an OCI reference for a file and a tag.
+func NewReference(file, tag string) (types.ImageReference, error) {
+	resolved, err := explicitfilepath.ResolvePathToFullyExplicit(file)
+	if err != nil {
+		return nil, err
+	}
+	// This is necessary to prevent file paths returned by PolicyConfigurationNamespaces
+	// from being ambiguous with values of PolicyConfigurationIdentity.
+	if strings.Contains(resolved, ":") {
+		return nil, errors.Errorf("Invalid OCI reference %s:%s: path %s contains a colon", file, tag, resolved)
+	}
+	if !refRegexp.MatchString(tag) {
+		return nil, errors.Errorf("Invalid tag %s", tag)
+	}
+
+	return ociArchiveReference{resolvedFile: resolved, tag: tag}, nil
+}
+
+// createOciRef creates the oci reference of the image
+func createOciRef(ref *ociArchiveReference) error {
+	// dir is a temporary directory used to generate the OCI image
+	dir, err := ioutil.TempDir("/var/tmp", "oci")
+	if err != nil {
+		return errors.Errorf("error creating file %v", err)
+	}
+	ociRef, err := ocilayout.NewReference(dir, ref.tag)
+	if err != nil {
+		return err
+	}
+	ref.ociRef = ociRef
+	ref.tempDirectory = dir
+	return nil
+}
+
+func (ref ociArchiveReference) Transport() types.ImageTransport {
+	return Transport
+}
+
+// StringWithinTransport returns a string representation of the reference, which MUST be such that
+// reference.Transport().ParseReference(reference.StringWithinTransport()) returns an equivalent reference.
+func (ref ociArchiveReference) StringWithinTransport() string {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
+			logrus.Error("error creating oci reference", err)
+		}
+	}
+	return ref.ociRef.StringWithinTransport()
+}
+
+// DockerReference returns a Docker reference associated with this reference
+func (ref ociArchiveReference) DockerReference() reference.Named {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
+			logrus.Error("error creating oci reference", err)
+		}
+	}
+	return ref.ociRef.DockerReference()
+}
+
+// PolicyConfigurationIdentity returns a string representation of the reference, suitable for policy lookup.
+func (ref ociArchiveReference) PolicyConfigurationIdentity() string {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
+			logrus.Error("error creating oci reference", err)
+		}
+	}
+	return ref.ociRef.PolicyConfigurationIdentity()
+}
+
+// PolicyConfigurationNamespaces returns a list of other policy configuration namespaces to search
+// for if explicit configuration for PolicyConfigurationIdentity() is not set
+func (ref ociArchiveReference) PolicyConfigurationNamespaces() []string {
+	if ref.ociRef == nil {
+		if err := createOciRef(&ref); err != nil {
+			logrus.Error("error creating oci reference", err)
+		}
+	}
+	return ref.ociRef.PolicyConfigurationNamespaces()
```

Comment:
Should not return a path to the temporary dir — and should not be modifying the FS here.